### PR TITLE
creating folder dialog box on pressing Enter/return key (#7021)

### LINKED ISSUE #7021: Enhancement: Enable submitting on the create folder dialog box on pressing Enter/Return key
**Description:**

The Create folder dialog box on the create new app page does not submit on pressing enter key on the keyboard, and can only be submitted by clicking the Create folder button.

**Steps to Reproduce:**

1. Open ToolJet and you will be presented with the create new app page.
2. In the sidebar just below the create new app button lies the section of folders.
3. Click on the **+** icon on the side of the heading **Folders.**
4. A **Create folder** dialog box opens up write a folder name and try to create it by pressing the enter/return key

**Expected Output:**

The create folder dialog box should submit and create folder when the user enters folder name and presses enter/return on the keyboard

https://github.com/ToolJet/ToolJet/assets/85864291/0ebbd375-3f45-4c68-bdd6-10bd8d0a9b62

### PR DESCRIPTION
**Pull request by FaroqueAbdullah**
Fixes #7021

### CODE REVIEW

File: frontend/src/HomePage/Folders.jsx [hunk 1]
```diff
@@ -317,6 +326,7 @@ export const Folders = function Folders({
               placeholder={t('homePage.foldersSection.folderName', 'folder name')}
               disabled={isCreating || isUpdating}
               value={newFolderName}
+              onKeyDown={(e) => e.key === 'Enter' && handleOnKeySubmit()}
```
Comment (Ckbhatia) [inline frontend/src/HomePage/Folders.jsx:329]: Simple function execution isn't costly. You can simply move the condition check inside this function.

**Review by Ckbhatia — COMMENTED** [review]
Optional small change requested

**Comment by FaroqueAbdullah** [comment]
Hi @Ckbhatia . Hope I have understood you properly. Please check and let me know.

File: frontend/src/HomePage/Folders.jsx [hunk 2]
```diff
@@ -317,6 +318,7 @@ export const Folders = function Folders({
               placeholder={t('homePage.foldersSection.folderName', 'folder name')}
               disabled={isCreating || isUpdating}
               value={newFolderName}
+              onKeyDown={(e) => (e.key === 'Enter' ? (showUpdateForm ? executeEditFolder() : saveFolder()) : null)}
```
Comment (nakulnagargade) [inline frontend/src/HomePage/Folders.jsx:321]: I think having a function definition is better than having two layers of ternary operators in JSX. Also will make our JSX clean and more readable.

Const handleFolderInputKeyDown = (e) => {
if (e.key === 'Enter') {
showUpdateForm ? executeEditFolder() : saveFolder()
}}

Comment (FaroqueAbdullah) [reply frontend/src/HomePage/Folders.jsx:321]: Not sure what is the best code format for this small task based on previous review

Comment (FaroqueAbdullah) [reply frontend/src/HomePage/Folders.jsx]: updated according to your review.

**Review by nakulnagargade — APPROVED** [review]

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**